packages/compass/src/app/components/update-toasts.tsx

```diff
@@ -114,6 +116,8 @@ export function onAutoupdateSuccess({
   onUpdate: () => void;
   onDismiss: () => void;
 }) {
+  // Before showing this toast, if the user has not dismissed the downloading toast, close it
+  closeToast(updateDownloadingToastId);
```

Comment:
once compass download finishes and before showing other toast to restart, closed the exiting update toast.

Reply:
We rarely need to manage toasts like that I think. If all toasts are representing the same state flow, they should just have the same id: then the toast logic will take care of closing / replacing the content with the correct one

Reply:
well in that case, i'd revert this bit

Reply:
To clarify, there is indeed an issue here, but instead of adding more manual cleanup everywhere, we should just use the same id for all the toasts here (like `compass-update` instead of `compass-update-{action}`): they are all triggered by one flow and we never expect to have multiple of those on the screen. Does this make sense?

Reply:
I was about to initiate a thread on design channel about this, but thank you for this comment. having same id solves this nicely. i added that in [1536537](https://github.com/mongodb-js/compass/pull/5768/commits/15365378839c0e67177ce18338f90d91d6b42275)